File: sdk/instructs/requests_instructs.py
import requests
import json
def send_api_request(instruction_set:list, 
                     session_id:str,
                     global_params:dict = {},
                     ModName = 'free_other',ApiName = 'instruct',
                     server_address = None,
                     ):
    '''
    Send request to mod with instrumentsAPI
    >>> global_params = {'参数名':'参数值'} # 注意是指令全局参数
    '''
    # 将全局指令参数添加到每个指令的instruction_parameters字段中
    if (not server_address):
        raise Exception('未知服务器地址')
    for instruction in instruction_set:
        if type(instruction) == str:
            break
        instruction['instruction_parameters'].update(global_params)

    params = {
        'ModName': ModName,
        'ApiName': ApiName,
        'instruction_set': json.dumps(instruction_set),
        'SessionID': session_id
    }
    # 发送请求
    response = requests.post(f'{server_address}/api/mod', params=params)

    # 处理响应
    if response.status_code == 200:
        # 解析并返回响应数据
        return response.json()
    else:
        # 处理请求错误
        return f'E url:{response.url}'
    

r =  send_api_request(session_id='1eaad741-93bc-4ff9-82da-97fa3733b4a4',
                 ModName='plugin_market',
                 instruction_set=[
                     {"instruction_name":"new_plugin","instruction_parameters":{
                         'github_publishing':'https://github.com/MCSLTeam/Plugins_is_Star_weaving_agent_download/releases/download/v1.0/Star_weaving_agent_download.rar',
                         'readme':'插件的一个例子（星织写的），本插件用于代理下载文件',
                         'Cloth':'https://www.df100.ltd/portrait.php',
                         'config':'''
{
    "plugin_name": "Star_weaving_agent_download",
    "version": "1.0",
    "description": "星织代理下载",
    "icon": "Cloth.png",
    "author": "星织",
    "author_email": "rc@163.com",
    "on_new_thread": true
  }''' 
                     }}
                 ]
                 ,server_address='http://127.0.0.1:11429')

print(r)

File: sdk/instructs/requests_instructs.js
function sendInstructsApiRequest(instructionSet, sessionId, globalParams, ModName = 'free_other', ApiName = 'instruct') {
    // Send request to mod with instrumentsAPI
    // globalParams should be a dictionary of global instruction parameters
    // Add global instruction parameters to the instruction_parameters field of each instruction
    instructionSet.forEach(instruction => {
      instruction.instruction_parameters = { ...instruction.instruction_parameters, ...globalParams };
    });
  
    const params = {
      ModName: ModName,
      ApiName: ApiName,
      instruction_set: JSON.stringify(instructionSet),
      SessionID: sessionId
    };
  
    // Send request
    fetch('/api/mod', { method: 'POST', body: new URLSearchParams(params) })
      .then(response => {
        if (response.ok) {
          // Parse and return response data
          return response.json();
        } else {
          // Handle request error
          throw new Error('Request failed');
        }
      })
      .catch(error => {
        // Handle error
        console.error(error);
        return null;
      });
  }

// import axios from 'https://cdn.skypack.dev/axios';

// async function sendApiRequest(instructionSet, sessionId, globalParams, ModName = 'free_other', ApiName = 'instruct') {
//     // 将全局指令参数添加到每个指令的instruction_parameters字段中
//     instructionSet.forEach(instruction => {
//       instruction['instruction_parameters'] = { ...instruction['instruction_parameters'], ...globalParams };
//     });
  
//     const params = {
//       ModName: ModName,
//       ApiName: ApiName,
//       instruction_set: JSON.stringify(instructionSet),
//       SessionID: sessionId
//     };
  
//     try {
//       // 发送请求
//       const response = await axios.post('/api/mod', { params });
  
//       // 处理响应
//       if (response.status === 200) {
//         // 解析并返回响应数据
//         return response.data;
//       } else {
//         // 处理请求错误
//         return null;
//       }
//     } catch (error) {
//       // 处理请求错误
//       return null;
//     }
//   }